proposals/cli_schema.md

```diff
@@ -0,0 +1,248 @@
+  - /fully/qualified/path/to/next/policy.yaml
+  resources:
+  - resource.yaml
+  - relative/path/to/next/resource.yaml
+  - /fully/qualified/path/to/next/resource.yaml
+  results:
+  # results[] vary depending on the rule type being tested.
+  # validate
+  - policy: policyname01 # required
+    rule: validaterulename # required
+    # resources must support multiple resources, all of which
+    # should be subject to the rest of the field values 
+    # in this result. (https://github.com/kyverno/kyverno/issues/2857)
+    resources: # required
+    - resource01
+    - resource02
+    - resource03
+    # namespace should set the namespace name for the resource(s), if not present,
+    # but otherwise should override the Namespace name if present. If the policy
+    # being tested is a `Policy` (Namespaced), and if the namespace is not defined
+    # in the resource AND not defined here, it should assume the value of `default`
+    # to be consistent with how Kubernetes treats resources (i.e., creating a resource
+    # with no metadata.namespace field implies the default namespace) (https://github.com/kyverno/kyverno/issues/4059)
+    # (https://github.com/kyverno/kyverno/issues/4087)
+    namespace: foonamespace # optional
+    kind: Kind # required. Should support values in kind and GVK formats.
+    result: myresult # required
+  # mutate
+  - policy: nextname
+    rule: mutaterulename
+    resources:
+    - resource01
+    patchedResource: mypatchedresource.yaml
+    kind: Kind
+    result: myresult
+  # generate
+  - policy: nextname
+    rule: generaterulename
+    resources:
+    - resource01
+    # for generate rules which use the `clone` directive, they must specify the `cloneSourceResource` field.
+    cloneSourceResource: clonesource.yaml
+    generatedResource: mypatchedresource.yaml
+    kind: Kind
+    result: myresult
+  # verifyImages. Must support attestations (https://github.com/kyverno/kyverno/issues/2866)
+  - policy: verify-attestations01
+    rule: verifyimagesrulename
+    resources:
+    - resource01
+    kind: Kind
+    result: myresult
+  variables:
+    # values that are in the global object apply to all policies, rules, and resources
+    # being tested. If a variable of the same name is found elsewhere (at a more granular level)
+    # its value at that level should trump the global value. The same rules apply for values
+    # here as elsewhere. See the variables.policies[].rules[].values for examples and comments.
+    # The hierarchy should be global => rule => resource.
+    global:
+      request.operation: CREATE
+    # policies[] is an array of objects. The name must be found in those specified in the files
+    # named in spec.resources[].
+    policies:
+    - name: policyname01
+      rules:
+      # the rule name must correspond to a rule found in the parent policy.
+      - name: rulename01
+        # values themselves do not need to be validated other than for YAML syntax
+        values:
```

Comment:
we have separate variables for foreach policies which is **foreachValues**
https://github.com/kyverno/kyverno/blob/56c90fd08717ca7184d1b5540bfd36906f9379d6/test/cli/test/context-foreach/values.yaml#L5

Reply:
Ah, another undocumented feature :) I didn't know about this.